templates/problem/voting-controls.html

```diff
@@ -0,0 +1,218 @@
+        <div id="id_has_voted_prompt">
+            <span class="vote-form-info">{{ _('Current Vote:') }}</span>
+            <span id="id_current_vote_value" class="vote-form-value"></span>
+            <br/>
+            <br/>
+            <span class="vote-form-text">{{  _('Change your vote below!') }}</span>
+            <br/>
+            <br/>
+        </div>
+        <div id="id_has_not_voted_prompt">
+            <span class="vote-form-text">{{ _("Cast your vote on this problem's weighting below!") }}</span>
+            <br/>
+            <br/>
+        </div>
+        <form id="id_delete_vote_form">
+            {% csrf_token %}
+        </form>
+        <form id="id_vote_form">
+            {% csrf_token %}
+            <span class="vote-form-info">{{ _('Points') }}</span>
+            <input id="id_points" class="vote-form-value" type="number" step="1"
+                   min="{{ min_possible_vote }}" max="{{ max_possible_vote }}" name="points">
+            <br/>
+            <div id="id_points_error_box">
+                <br/>
+                <span id="id_points_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <span class="vote-form-info">{{ _('Note') }}</span>
+            <br/>
+            <textarea id="id_note" class="vote-form-text" name="note" rows="6" cols="80"></textarea>
+            <br/>
+            <div id="id_note_error_box">
+                <br/>
+                <span id="id_note_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <div>
+                <input type="submit" id="id_vote_form_submit_button" class="vote-button" value="{{ _('Vote') }}">
+                <input type="submit" id="id_delete_vote_form_submit_button" value="{{ _('Delete vote') }}" form="id_delete_vote_form">
+            </div>
+        </form>
+    </div>
+
+    <div>
+        <a class="form-button" id="id_vote_button"></a>
+    </div>
+
+    {% if vote is not none %}
+        {{ vote.note|json_script('voted_note') }}
+    {% endif %}
+    <script>
+        let voted_points = null;
+        let voted_note = null;
+        {% if vote is not none %}
+            let has_voted = true;
+            voted_points = {{ vote.points }};
+            voted_note = JSON.parse(document.getElementById('voted_note').textContent);
+        {% else %}
+            let has_voted = false;
+        {% endif %}
+
+        function voteUpdate(){
+            $('#id_has_voted_prompt').show();
+            $('#id_current_vote_value').prop('innerText', voted_points);
+            $('#id_has_not_voted_prompt').hide();
+            $('#id_delete_vote_form_submit_button').show();
+
+            $('#id_vote_button').prop('innerText', `{{ _('Change vote') }} (` + voted_points + ')');
+            $('#id_points').prop('value', '');
+            $('#id_points').prop('placeholder', voted_points);
```

Comment:
Don't use placeholder here, use something like `value` so that the user doesn't have to re-enter the point value if they want to change the reasoning.

Reply:
Fixed.